go/arrow/array/concat_test.go

```diff
@@ -568,3 +569,59 @@ func TestConcatDictionaryNullSlots(t *testing.T) {
 
 	assert.Truef(t, array.Equal(actual, expected), "got: %s, expected: %s", actual, expected)
 }
+
+func TestConcatRunLengthEncoded(t *testing.T) {
```

Comment:
I suppose:

- Can we test all three index types?
- Can we test that we error properly on overflow?
- Can we test that we can concatenate right up to the edge of overflow?

Reply:
added